state-proofs/sui_state_proof/sources/sui_state_proof.move

```diff
@@ -0,0 +1,37 @@
+module sui_state_proof::sui_state_proof {
+    use std::vector;
+    use sui::object::{Self, ID, UID};
+    use sui::event;
+    use sui::tx_context::TxContext;
+
+
+    struct DWalletCap has key, store {
```

Comment:
The fact there is no backward link here, and that there is just a one-way link from dWallet network here and not the opposite, means that two different dWallets could be controlled by the same capability.

This might not be a problem, but is a conscious design decision that needs to be made intentionally, and not as an unintentional byproduct.

I can see a reason why we'd want two different dWallets to be controlled by the same sui module, but I'm not sure that this is the right way to do it: instead, a single object could wrap/hold as child objects two different capabilities. IMO this is clearer, and it achieves the same objective.

@omersadika thoughts?